src/ol/source/tileimage.js

```diff
@@ -278,6 +278,84 @@ ol.source.TileImage.prototype.getTile = function(z, x, y, pixelRatio, projection
   }
 };
 
+/**
+ * Get the interim tile most suitable for rendering using the chain of interim tiles.
+ * This corresponds to the loaded tile with the highest sequence, if no such
+ * tile exists, the original tile is returned.
+ * @param {ol.Tile} tile Most recent tile (head of the interim tile list)
+ * @return {!ol.Tile} Best tile for rendering at this location.
+ */
+ol.source.TileImage.prototype.getInterimTile = function(tile) {
+  var head = tile;
+  if (!head.interimTile) {
+    //empty chain
+    return head;
+  }
+  tile = head.interimTile;
+
+  //find the first loaded tile and return it. Since the chain is sorted in decreasing
+  //sequence number, there is no need to search the remaineder of the list (all those
+  //tiles correspond to older requests and will be cleaned up by refreshInterimChain)
+  do {
+    if (tile.getState() == ol.Tile.State.LOADED) {
+      return tile;
+    }
+    tile = tile.interimTile;
+  } while (tile);
+
+  //we can not find a tile better than the head
+  return head;
+};
+
+/**
+ * Goes through the chain of interim tiles starting at the provided tile and discards
+ * sections of the chain that are no longer relevant.
+ * @param {ol.Tile} tile Head of chain of interim tiles.
+ * @return {void}
+ * @private
+ */
+ol.source.TileImage.prototype.refreshInterimChain = function(tile) {
+  var /** @type {ol.Tile} */ head = tile;
+
+  if (!head.interimTile) {
+    return;
+  }
+
+  tile = head.interimTile;
+  var prev = head;
+
+  var discard = function(prev, tile) {
+    prev.interimTile = tile.interimTile;
+    return prev;
+  };
+  var step = function(prev,tile) {
```

Comment:
I like your update to the linked list traversal stuff, it certainly saves some code.